source/Irrlicht/CBAWMeshFileLoader.cpp

```diff
@@ -0,0 +1,206 @@
+#include "CBAWMeshFileLoader.h"
+
+#include "CFinalBoneHierarchy.h"
+#include "SMesh.h"
+#include "CSkinnedMesh.h"
+
+namespace irr { namespace scene
+{
+CBAWMeshFileLoader::~CBAWMeshFileLoader()
+{
+	if (m_fileSystem)
+		m_fileSystem->drop();
+}
+
+CBAWMeshFileLoader::CBAWMeshFileLoader(scene::ISceneManager* _sm, io::IFileSystem* _fs) : m_sceneMgr(_sm), m_fileSystem(_fs)
+{
+#ifdef _DEBUG
+	setDebugName("CBAWMeshFileLoader");
+#endif
+	if (m_fileSystem)
+		m_fileSystem->grab();
+}
+
+template<>
+core::ICPUBuffer* CBAWMeshFileLoader::make<core::ICPUBuffer>(SPair& _data, SContext&) const
+{
+	typename core::CorrespondingBlobTypeFor<core::ICPUBuffer>::type* blob = core::toBlobPtr<core::ICPUBuffer>(_data.blob);
+
+	void* blobData = malloc(_data.header->blobSizeDecompr);
+	memcpy(blobData, blob->getData(), _data.header->blobSizeDecompr);
+	return new core::ICPUBuffer(_data.header->blobSizeDecompr, blobData);
+}
+template<>
+IMeshDataFormatDesc<core::ICPUBuffer>* CBAWMeshFileLoader::make<IMeshDataFormatDesc<core::ICPUBuffer> >(SPair& _data, SContext& _ctx) const
+{
+	typename core::CorrespondingBlobTypeFor<IMeshDataFormatDesc<core::ICPUBuffer> >::type* blob = core::toBlobPtr<IMeshDataFormatDesc<core::ICPUBuffer> >(_data.blob);
+
+	IMeshDataFormatDesc<core::ICPUBuffer>* desc = new scene::ICPUMeshDataFormatDesc();
+	for (E_VERTEX_ATTRIBUTE_ID i = EVAI_ATTR0; i < EVAI_COUNT; i = E_VERTEX_ATTRIBUTE_ID((int)i + 1))
+	{
+		if (blob->attrBufPtrs[(int)i])
+			desc->mapVertexAttrBuffer(
+				make<core::ICPUBuffer>(_ctx.blobs[blob->attrBufPtrs[(int)i]], _ctx),
+				i,
+				blob->cpa[(int)i],
+				blob->attrType[(int)i],
+				blob->attrStride[(int)i],
+				blob->attrOffset[(int)i],
+				blob->attrDivisor[(int)i]
+			);
+	}
+	if (blob->idxBufPtr)
+		desc->mapIndexBuffer(make<core::ICPUBuffer>(_ctx.blobs[blob->idxBufPtr], _ctx));
+
+	return desc;
+}
+template<>
+video::IVirtualTexture* CBAWMeshFileLoader::make<video::IVirtualTexture>(SPair& _data, SContext& _ctx) const
+{
+	typename core::CorrespondingBlobTypeFor<video::IVirtualTexture>::type* blob = core::toBlobPtr<video::IVirtualTexture>(_data.blob);
+
+	video::ITexture* texture;
+	const char* const texname = (const char*)blob->getData();
+	bool newTexture;
+	if (m_fileSystem->existFile(texname))
+	{
+		newTexture = m_sceneMgr->getVideoDriver()->findTexture(texname) == 0;
+		texture = m_sceneMgr->getVideoDriver()->getTexture(texname);
+	}
+	else
+	{
+		const io::path path = _ctx.filePath + texname;
+		newTexture = m_sceneMgr->getVideoDriver()->findTexture(path) == 0;
+		// try to read from the path relative to where the .baw is loaded from
+		texture = m_sceneMgr->getVideoDriver()->getTexture(path);
+	}
+	//! @todo @bug Do somemthing with `newTexture`? In obj loader something happens in case where newTexture is true
```

Comment:
What about this? Don't really know why anything would be needed, but .obj loader does something...

Reply:
I'll test next week, and see